### DISCUSSION TITLE
Mime types and reject exe file

### DISCUSSION THREAD

**Discussion by TeXmeX66** [rspamd/rspamd#5076] [General]
Hello,
I have a configuration problem with Rspamd. I am trying to ensure that emails containing files with '.exe' attachments or within archives are rejected. In the file /etc/rspamd/local.d/mime_types.conf, I have set it up as follows 

```
bad_extensions = {
  scr = 4,
  lnk = 4,
  exe = 100,
  jar = 2,
  com = 4,
  bat = 4,
  ace = 4,
  arj = 4,
  cab = 3,
};

bad_archive_extensions = {
  pptx = 0.1,
  docx = 0.1,
  xlsx = 0.1,
  pdf = 0.1,
  jar = 3,
  js = 4,
  vbs = 4,
  exe = 100,
};
```

 I sent an exe file in an email, but the email is not being rejected. Here is what I get in the log: _MIME_BAD_EXTENSION (10.1) [exe]_. The total score is 12.10 / 25. 

Where is the error, please? 

Thank you.

**Comment by moisseev** [comment]
The issue is that the **max_score** setting in `scores.d/mime_types_group.conf` is set to `10.0`. As a result, even though you have configured high scores for .exe files, the score might not be high enough to trigger the rejection.

To resolve this, you should create a local configuration file to override the `max_score` setting:

local.d/mime_types_group.conf:

```
max_score = 200.0;  # Increase this value as needed
```

**Reply by vstakhov** [reply]
Side thought: maybe we can set max_score as a percentage of the reject action threshold? Will it be useful?

**Reply by moisseev** [reply]
I'm not sure this approach would be effective. The default Out-Of-Box configuration is generally well-balanced with respect to scores, thresholds, and limits. Altering these thresholds could disrupt this balance and lead to unintended consequences. A better approach might be to adjust individual symbol scores instead of the thresholds. Specifically, setting `max_score` as a percentage of the reject action threshold might create inconsistencies between individual scores, group scores, and other thresholds.

In the specific scenario mentioned by the OP, adjusting `max_score` to a percentage of the reject threshold won't be beneficial. For instance, if you set `max_score` to 0.67, and the reject threshold is also set at 0.67, the reject action won’t be triggered because the score will still be below the threshold.
